### PR TITLE
DOC: update the pandas.DataFrame.pct_change docstring

### PR DESCRIPTION
**Pull request by myles**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ ] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
################### Docstring (pandas.DataFrame.pct_change)  ###################
################################################################################

Percentage change between the current and previous element.

This is useful in comparing the percentage of change in a time series
of elements.

Parameters
----------
periods : int, default 1
    Periods to shift for forming percent change.
fill_method : str, default 'pad'
    How to handle NAs before computing percent changes.
limit : int, default None
    The number of consecutive NAs to fill before stopping.
freq : DateOffset, timedelta, or offset alias string, optional
    Increment to use from time series API (e.g. 'M' or BDay()).
**kwargs : mapping, optional
    Additional keyword arguments are passed into
    `DataFrame.shift` or `Series.shift`.

Returns
-------
chg : Series or DataFrame
    The same type as the calling object.

Examples
--------
See the percentage change in a Series.

>>> s = pd.Series([90, 91, 85])
>>> s
0    90
1    91
2    85
dtype: int64
>>> s.pct_change()
0         NaN
1    0.011111
2   -0.065934
dtype: float64

See the percentage change in a Series where filling NAs with last
valid observation forward to next valid.

>>> s = pd.Series([90, 91, None, 85])
>>> s
0    90.0
1    91.0
2     NaN
3    85.0
dtype: float64
>>> s.pct_change(fill_method='ffill')
0         NaN
1    0.011111
2    0.000000
3   -0.065934
dtype: float64

Percentage change in French franc, Deutsche Mark, and Italian lira from
1 January 1980 to 1 March 1980.

>>> df = pd.DataFrame({
...     'FR': [4.0405, 4.0963, 4.3149],
...     'GR': [1.7246, 1.7482, 1.8519],
...     'IT': [804.74, 810.01, 860.13]},
...     index=['1980-01-01', '1980-02-01', '1980-03-01'])
>>> df
                FR      GR      IT
1980-01-01  4.0405  1.7246  804.74
1980-02-01  4.0963  1.7482  810.01
1980-03-01  4.3149  1.8519  860.13
>>> df.pct_change()
                  FR        GR        IT
1980-01-01       NaN       NaN       NaN
1980-02-01  0.013810  0.013684  0.006549
1980-03-01  0.053365  0.059318  0.061876

Percentage of change in GOOG and APPL stock volume.

>>> df = pd.DataFrame({
...     '2016': [1769950, 30586265],
...     '2015': [1500923, 40912316],
...     '2014': [1371819, 41403351]},
...     index=['GOOG', 'APPL'])
>>> df
          2016      2015      2014
GOOG   1769950   1500923   1371819
APPL  30586265  40912316  41403351
>>> df.pct_change(axis='columns')
      2016      2015      2014
GOOG   NaN -0.151997 -0.086016
APPL   NaN  0.337604  0.012002

See Also
--------
DataFrame.diff : see the difference of two columns
Series.diff : see the difference of two columns

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Errors in parameters section
		Parameters {'kwargs'} not documented
		Unknown parameters {'**kwargs'}
```

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
...
         fill_method : str, default 'pad'
-            How to handle NAs before computing percent changes
+            How to handle NAs before computing percent changes.
         limit : int, default None
-            The number of consecutive NAs to fill before stopping
+            The number of consecutive NAs to fill before stopping.
         freq : DateOffset, timedelta, or offset alias string, optional
-            Increment to use from time series API (e.g. 'M' or BDay())
+            Increment to use from time series API (e.g. 'M' or BDay()).
+        kwargs : mapping, optional
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: `kwargs : mapping, optional` -> `**kwargs`

File: pandas/core/generic.py [hunk 2]
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
...
+            A dictionary of keyword arguments passed into
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: It's not a dictionary, but just additional keywords, so you cna make this ""Additional keyword arguments are passed .."

Comment (jorisvandenbossche) [reply pandas/core/generic.py]: But on second thought, are there actually additional keywords for shift?

File: pandas/core/generic.py [hunk 3]
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
...
+            ``DataFrame.shift``/``Series.shift``.
 
         Returns
         -------
-        chg : %(klass)s
+        chg : Series or DataFrame, same type as the input
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: 'input' -> 'calling object'

File: pandas/core/generic.py [hunk 4]
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
...
+        >>> df.pct_change()
+                          FR        GR        IT
+        1980-01-01       NaN       NaN       NaN
+        1980-02-01  0.013810  0.013684  0.006549
+        1980-03-01  0.053365  0.059318  0.061876
+
+        See Also
+        --------
+        pandas.DataFrame.diff : see the difference of two columns
+        pandas.Series.diff : see the difference of two columns
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: Can you move this above the "Examples" section?

**Review by jorisvandenbossche — COMMENTED** [review]
Nice PR! Added a few comments

File: pandas/core/generic.py (hunk 2 above)
Comment (myles) [reply pandas/core/generic.py]: That's what it's mapped to https://github.com/pandas-dev/pandas/blob/7169830040255c1073f99e2ee8817596a67b1cd5/pandas/core/generic.py#L7561-L7562

File: pandas/core/generic.py [hunk 5]
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
...
         fill_method : str, default 'pad'
-            How to handle NAs before computing percent changes
+            How to handle NAs before computing percent changes.
         limit : int, default None
-            The number of consecutive NAs to fill before stopping
+            The number of consecutive NAs to fill before stopping.
         freq : DateOffset, timedelta, or offset alias string, optional
-            Increment to use from time series API (e.g. 'M' or BDay())
+            Increment to use from time series API (e.g. 'M' or BDay()).
+        **kwargs : mapping, optional
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Can remove the type here. Just `**kwargs`.

File: pandas/core/generic.py [hunk 6]
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
...
+            Additional keyword arguments are passed into
+            ``DataFrame.shift``/``Series.shift``.
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Single backticks. "or" instead of `/`

File: pandas/core/generic.py [hunk 7]
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
...
 
         Returns
         -------
-        chg : %(klass)s
+        chg : Series or DataFrame, same type as the calling object
```
Comment (TomAugspurger) [inline pandas/core/generic.py]:
```
change : Series or DataFrame
    The same type as the calling object.
```

File: pandas/core/generic.py [hunk 8]
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
...
 
         Notes
         -----
-
         By default, the percentage change is calculated along the stat
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: I don't think most users will understand `stat` access. Best to remove this and only focus on Series / DataFrame.

File: pandas/core/generic.py [hunk 9]
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
...
         axis: 0, or ``Index``, for ``DataFrame`` and 1, or ``minor`` for
         ``Panel``. You can change this with the ``axis`` keyword argument.
+
+        Percentage change in French franc, Deutsche Mark, and Italian lira from
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Move under the "Examples" header.

File: pandas/core/generic.py [hunk 10]
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
...
+                        FR      GR      IT
+        1980-01-01  4.0405  1.7246  804.74
+        1980-02-01  4.0963  1.7482  810.01
+        1980-03-01  4.3149  1.8519  860.13
+        >>> df.pct_change()
+                          FR        GR        IT
+        1980-01-01       NaN       NaN       NaN
+        1980-02-01  0.013810  0.013684  0.006549
+        1980-03-01  0.053365  0.059318  0.061876
+
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: Show an example with `axis='columns'`. Can use the same DataFrame, even though it's a silly example for axis=1.

File: pandas/core/generic.py (hunk 4 above)
Comment (TomAugspurger) [reply pandas/core/generic.py]: Can also remove the `pandas.` prefix I think.

**Comment by TomAugspurger** [comment]
Made some touchups. Thanks @myles!

**Comment by jorisvandenbossche** [comment]
@TomAugspurger small comment on when merging: can you remove all the different commit messages (in case of those DOC PRs it's mostly useless to keep them). It's maybe not that important and only my nitpicky self, but it's only a "click in box, ctrl-A, del" combo away :-)

**Comment by TomAugspurger** [comment]
👍

**Pull request merged**